### PR TITLE
fix: invoke onProgress callback with DHT queries during routing

### PR DESCRIPTION
**Pull request by achingbrain**
Allow passing an `onProgress` callback to the peer/content routers that can receive DHT query events.

Refs #1574

### CODE REVIEW

**Review by maschad — APPROVED** [review]
ProgessEvents! Woot!

File: test/content-routing/content-routing.node.ts [hunk 1]
```diff
@@ -115,6 +115,39 @@ describe('content-routing', () => {
...
+            multiaddrs: [],
+            protocols: []
+          }]
+        }
+        deferred.resolve()
+      })
+
+      const onProgress = sinon.stub()
+
+      await drain(nodes[0].contentRouting.findProviders(CID.parse('QmU621oD8AhHw6t25vVyfYKmL9VV3PTgc52FngEhTGACFB'), {
```
Comment (maschad) [inline test/content-routing/content-routing.node.ts:142]: Where did this _CID_ come from?

Comment (achingbrain) [reply test/content-routing/content-routing.node.ts:142]: It's used elsewhere in the test suite 🤷

**Comment by achingbrain** [comment]
Converted to draft because we need to be a bit cleverer about how we expose the `ProgressEvent` types.

If we don't expose them all we get errors:

```js
await name.resolve(key, {
  onProgress(evt) {
    if (evt.type === 'content-routing:get:dht:event') {

    }
  }
})
```

```
This comparison appears to be unintentional because the types '"ipns:resolve:start" |
"ipns:resolve:success" | ..(more events here).. and '"content-routing:get:dht:event"'
have no overlap.
```

The problem is we want to avoid exposing all the ProgressEvent types for all the content routing/peer routing implementations at the content routing/peer routing interface level.  Better to somehow aggregate the types based on the configuration passed to the libp2p factory function in a similar way to #1563

**Comment by achingbrain** [comment]
Closing in favour of #1975

**Pull request closed without merging**